### PR TITLE
Remove SystemKeyspace from EngineTxFactory

### PR DESCRIPTION
**Pull request by marco-scoppetta**
# Why is this PR needed?

So that we inject the `GraknKeyspaceStore` (previously called `SystemKeyspace`) down to Engine.
This way we can customise the way we access the `graknsystem` keyspace.
This is necessary because `GraknKeyspaceStore` is the only part in Grakn where we access the database without a direct input from the user, therefore this was causing problem when enabling authentication on the database.

# What does the PR do?

 rename `SystemKeyspace` to `GraknKeyspaceStore`, given that this class actually represents a set of available keyspaces.

Small refactoring to code an tests - nothing major

# Does it break backwards compatibility?

No
# List of future improvements not on this PR

A couple of TODOs

### CODE REVIEW

File: grakn-engine/src/main/java/ai/grakn/engine/GraknKeyspaceStoreImpl.java [hunk 1]
```diff
@@ -108,19 +80,25 @@ private void logNewKeyspace(Keyspace keyspace){
...
-        return "/creating-new-keyspace-lock/" + keyspace.getValue();
-    }
-
     @Override
     public boolean containsKeyspace(Keyspace keyspace){
-        try (GraknTx graph = factory.tx(SYSTEM_KB_KEYSPACE, GraknTxType.READ)) {
-            return graph.getAttributeType(KEYSPACE_RESOURCE.getValue()).getAttribute(keyspace) != null;
+        //Check the local cache to see which keyspaces we already have open
+        if(existingKeyspaces.contains(keyspace)){
+            return true;
```
Comment (aelred) [inline grakn-engine/src/main/java/ai/grakn/engine/GraknKeyspaceStoreImpl.java]: 👍

File: grakn-engine/src/main/java/ai/grakn/engine/SystemKeyspaceSessionProvider.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import ai.grakn.kb.internal.EmbeddedGraknTx;
+
+/**
+ * Provides transactions to "graknsystem" which is Grakn system keyspace - used by GraknKeyspaceStore to keep track
+ * of available keyspaces
+ *
+ * @author Marco Scoppetta
+ */
+
+public class SystemKeyspaceSessionProvider implements SystemKeyspaceSession {
```
Comment (aelred) [inline grakn-engine/src/main/java/ai/grakn/engine/SystemKeyspaceSessionProvider.java]: Calling this `_Provider` confused me. I thought it was a factory :(

I don't know what a good name is though. `SystemKeyspaceSessionImpl` is a bit silly 🤷‍♂️

Comment (marco-scoppetta) [reply grakn-engine/src/main/java/ai/grakn/engine/SystemKeyspaceSessionProvider.java]: @aelred, maybe `GraknSystemKeyspaceSession` ?

Comment (aelred) [reply grakn-engine/src/main/java/ai/grakn/engine/SystemKeyspaceSessionProvider.java]: 👍 
Makes sense, and that's the convention you used before I think.

**Pull request merged**